libs/until-destroy/src/lib/checker.ts

```diff
@@ -60,6 +70,19 @@ export function setupSubjectUnsubscribedChecker(instance: any, destroy$: Subject
   instance[CheckerHasBeenSet] = true;
 }
 
+// We can't use `observeOn(asapScheduler)` because this might break the app's change detection.
+// RxJS schedulers coalesce tasks and then flush the queue, which means our task might be scheduled
+// within the root zone, and then all of the tasks (that were set up by developers in the Angular zone)
+// will also be flushed in the root zone.
+function observeOnPromise<T>() {
```

Comment:
You can use `mergeMap` instead of creating new observable as you're already imported it:

```ts
mergeMap((value) => Promise.resolve(value))
```